OpenRA.Mods.Common/Activities/Repair.cs

```diff
@@ -32,7 +33,12 @@ public override Activity Tick(Actor self)
 			if (health == null) return NextActivity;
 
 			if (health.DamageState == DamageState.Undamaged)
+			{
+				var repairsUnits = host.Info.Traits.Get<RepairsUnitsInfo>();
```

Comment:
You have the same thing at  https://github.com/abcdefg30/OpenRA/blob/repairnotifications/OpenRA.Mods.Common/Activities/Repair.cs#L45
Cache this in the constructor for best results.